app.py

```diff
@@ -19,27 +43,46 @@ def version_proxy():
 
 
 @app.route("/api/predict", methods=["POST"])
+@predict_latency.time()  # Measure request duration
 def predict_proxy():
     payload = request.get_json()
-    response = requests.post(f"{APP_SERVICE_URL}/api/predict", json=payload)
-    return jsonify(response.json()), response.status_code
+
+    try:
+        response = requests.post(f"{APP_SERVICE_URL}/api/predict", json=payload)
+        prediction_requests_total.labels(status=str(response.status_code)).inc()
+        return jsonify(response.json()), response.status_code
+    except Exception as e:
+        prediction_requests_total.labels(status="500").inc()
+        return jsonify({"error": str(e)}), 500
 
 
 @app.route("/api/feedback", methods=["POST"])
 def feedback_proxy():
     payload = request.get_json()
-    response = requests.post(f"{APP_SERVICE_URL}/api/feedback", json=payload)
-    return "", response.status_code
+    try:
+        is_correct = payload.get("is_correct")
+        if is_correct is True:
+            feedback_rating_total.labels(feedback_type="positive").inc()
+        elif is_correct is False:
+            feedback_rating_total.labels(feedback_type="negative").inc()
+        else:
+            feedback_rating_total.labels(feedback_type="unknown").inc()
+
+        response = requests.post(f"{APP_SERVICE_URL}/api/feedback", json=payload)
+        return "", response.status_code
+    except Exception as e:
+        return jsonify({"error": str(e)}), 500
 
 
 @app.route("/")
 def index():
+    active_users_total.inc()
     return render_template("index.html", app_service_url=APP_SERVICE_URL)
 
 
 if __name__ == "__main__":
     port = os.getenv("PORT", 3000)
-    debug = os.getenv("DEBUG", "false").lower() == "true"
+    debug = os.getenv("DEBUG", "false").lower() == "false"
```

Comment:
Do we intend for this to happen? 
This will basically always start the app in debug mode if we don't provide the DEBUG variable in the environment, which we aren't providing in production code in helm, if I remember correctly.

Reply:
When I was testing prometheus client I changed it and forgot to change back, good catch, thanks!